script.js

```diff
@@ -0,0 +1,194 @@
+    style: "currency",
+    currency: "RUB",
+    currencyDisplay: "symbol",
+    minimumFractionDigits: 0,
+    maximumFractionDigits: 0,
+  }).format(ticketData.price);
+  ticketLogo.src = carrierLogoUrl;
+
+  // формируем и заполняем маршруты билета
+  ticketData.segments.forEach((element) => {
+    const routeTemplate = ticket.querySelector("#route-template").content;
+    const ticketRoute = routeTemplate.cloneNode(true);
+    const routeTitle = ticketRoute.querySelector(".route-title");
+    const routeTime = ticketRoute.querySelector(".route-time");
+    const routeLenght = ticketRoute.querySelector(".route-lenght");
+    const routeSstopsCount = ticketRoute.querySelector(".route-stops-count");
+    const routeStops = ticketRoute.querySelector(".route-stops");
+
+    let routeDuration = {};
+    routeDuration.mins = element.duration % 60;
+    routeDuration.hours = (element.duration - routeDuration.mins) / 60;
+
+    // получаем дату и времы вылета из строки и вычисляем время прилета использую продолжительность полета (без учета часовых поясов)
+    let timeFormatOptions = { hour: "numeric", minute: "numeric" };
+    let departureDate = new Date(element.date);
+    let departureTime = departureDate.toLocaleTimeString(
+      "ru-RU",
+      timeFormatOptions
+    );
+    let arrivalDate = new Date(departureDate);
+    arrivalDate.setMinutes(arrivalDate.getMinutes() + element.duration);
+    let arrivalTime = arrivalDate.toLocaleTimeString(
+      "ru-RU",
+      timeFormatOptions
+    );
+
+    // заполняем маршрут
+    routeTitle.textContent = `${element.origin} – ${element.destination}`;
+    routeTime.textContent = `${departureTime} – ${arrivalTime}`;
+    routeLenght.textContent = `${routeDuration.hours}ч ${routeDuration.mins}м`;
+    routeSstopsCount.textContent = defineQuantityStops(element.stops.length);
+    routeStops.textContent = element.stops.join(", ");
+
+    ticket.children[0].append(ticketRoute); // добавляем в потомка DocumentFragment HTMLCollection
+  });
+
+  document.querySelector("#tickets-section").append(ticket); // добавляем заполненный экзепляр template в документ
+}
+
+// склоняем пересадку
+function defineQuantityStops(number) {
+  const words = ["без пересадок", "пересадка", "пересадки", "пересадок"];
+  let n =
+    (number >= 5 && number <= 20) ||
+    (number % 10 >= 5 && number % 10 <= 20) ||
+    (number != 0 && number % 10 == 0)
+      ? 3
+      : number % 10 == 1
+      ? 1
+      : number % 10 >= 2 && number % 10 <= 4
+      ? 2
+      : 0;
+  if (number == 0) {
+    return words[n];
+  } else {
+    return `${number} ${words[n]}`;
+  }
+}
+
+//удаляем все билеты
+function clearTickets() {
+  let ticketsSection = document.querySelector("#tickets-section");
+  while (ticketsSection.children.length > 1) {
+    ticketsSection.lastElementChild.remove();
+  }
+}
+
+// выбираем билеты по количеству пересадок
+function getFilteredTickets(ticketsArray, ...filters) {
+  if (filters.includes(-1)) {
```

Comment:
Для фиксированных значений следует использовать именованные константы. Это улучшит восприятие кода

```suggestion
  if (filters.includes(FILTER_OPTION_ALL)) {
```

Reply:
Константы в этом случае принято обозначать капсом:

```js
const FILTER_OPTION_ALL = -1;
const FILTER_OPTION_STOPS_0 = 0;
const FILTER_OPTION_STOPS_1 = 1;
const FILTER_OPTION_STOPS_2 = 1;
```

Reply:
Еще я рекомендую в составлять название таких констант по правилу: `const <КОНТЕКСТ>_<ЗНАЧЕНИЕ> = ...`